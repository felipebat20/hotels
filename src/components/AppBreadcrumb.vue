<template>
  <div>
    <q-breadcrumbs active-color="primary" separator=">" style="font-size: 16px">
      <template v-slot:separator>
        <q-icon
          name="chevron_right"
          color="primary"
          size="16px"
        />
      </template>

      <q-breadcrumbs-el
        v-for="breadcrumb in breadcrumbItems"
        :key="breadcrumb.label"
        :label="breadcrumb.label"
      />

    </q-breadcrumbs>
  </div>
</template>

<script setup lang="ts">
import { PropType } from 'vue';

defineProps({
  breadcrumbItems: {
    type: Array as PropType<{ label: string }[]>,
    required: true,
  },
});
</script>