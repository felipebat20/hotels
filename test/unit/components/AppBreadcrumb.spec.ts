import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-vitest';
import { mount } from '@vue/test-utils';
import { describe, expect, it, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing'

import AppBreadcrumb from '@/components/AppBreadcrumb.vue';

installQuasarPlugin();

describe('test AppBreadcrumb', () => {
  const createWrapper = () => {
    return mount(AppBreadcrumb, {
      props: {
        breadcrumbItems: [
          { label: 'Inicio' },
          { label: 'HotÃ©is' },
        ],
      },
      global: {
        plugins: [createTestingPinia({ createSpy: vi.fn })],
      },
    });
  };

  it('should mount component properly', () => {
    const wrapper = createWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
