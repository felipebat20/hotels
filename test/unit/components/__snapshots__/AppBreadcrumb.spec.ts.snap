// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test AppBreadcrumb > should mount component properly 1`] = `
"<div>
  <div class="q-breadcrumbs" style="font-size: 16px;">
    <div class="flex items-center justify-start q-gutter-sm">
      <div class="flex items-center text-primary"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">Inicio</span></div>
      <div class="q-breadcrumbs__separator"><i class="q-icon text-primary notranslate material-icons" style="font-size: 16px;" aria-hidden="true">chevron_right</i></div>
      <div class="flex items-center q-breadcrumbs--last"><span class="q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable">Hot√©is</span></div>
    </div>
  </div>
</div>"
`;
